src/clib/pio_darray.c

```diff
@@ -110,8 +115,11 @@ int PIOc_write_darray_multi(int ncid, const int *varids, int ioid, int nvars,
     var_desc_t *vdesc0;    /* Array of var_desc structure for each var. */
     int fndims;            /* Number of dims in the var in the file. */
     int mpierr = MPI_SUCCESS, mpierr2;  /* Return code from MPI function calls. */
-    int ierr;              /* Return code. */
+    int ierr = PIO_NOERR;              /* Return code. */
```

Comment:
Setting ierr here is incorrect.

If you examine the code more closely, you will see that ierr is already set everywhere that it needs to be set. Setting it as the beginning of the function signals that this is necessary (which it is in some cases, but not this function).

Setting of ierr, ret, mpierr, and other return values at the beginning of the function occurs whenever the return value may be set on some processors and not others. For example, if only the ioprocs set ierr, then it must be set to PIO_NOERR at the top of the function so that the computation procs have a good value.

But when it is set on all procs in the code, then it does not need to be set at the beginning of the function. That is the case with this function, so it is incorrect to set it here.